src/game/server/game.cpp

```diff
@@ -24,6 +24,28 @@ void MapCycleFileChangedCallback( IConVar *var, const char *pOldString, float fl
 	}
 }
 
+void flashlight_changed( IConVar *pConVar, const char *pOldString, float flOldValue )
+{
+	ConVarRef var( pConVar );
+	if ( !var.IsValid() )
+	{
+		return;
+	}
+
+	if ( var.GetInt() < 1 )
```

Comment:
Should be == 0 to match `CMultiplayRules::FAllowFlashlight` behavior, -1 will enable flashlights but your code will disable them